### ISSUE TITLE
Feature: Export to PDF/CSV

### ISSUE THREAD

**Issue by Tholkappiar** [ksauraj/jee_counsellor#19]
### Description
New feature that allows users to export the displayed data to PDF or CSV directly from the HTML page or form the script itself. This enhancement would provide users with the ability to save and share the information in a convenient and portable format.

Is this PDF/CSV formats or okay for the tool , or if you have any other format suggestions kidly give the response regarding this !

### Expected Behavior
- The export options should include PDF and CSV formats.
- The exported file should accurately represent the currently displayed data on the HTML page.

**Comment by ksauraj** [comment]
> ## Feature Request: Export to PDF/CSV
> ### Description
> New feature that allows users to export the displayed data to PDF or CSV directly from the HTML page or form the script itself. This enhancement would provide users with the ability to save and share the information in a convenient and portable format.
> 
> Is this PDF/CSV formats or okay for the tool , or if you have any other format suggestions kidly give the response regarding this !
> 
> ### Expected Behavior
> * The export options should include PDF and CSV formats.
> * The exported file should accurately represent the currently displayed data on the HTML page.

Seems useful and intestine feature to add, would you like to work on this?

**Comment by Tholkappiar** [comment]
yes ofcourse , can you provide the export format which will be useful for the users ?

**Comment by ksauraj** [comment]
> yes ofcourse , can you provide the export format which will be useful for the users ?

In my opinion, pdf will be better option. Also talking about csv it won't take much efforts as we already have option in `pandas` to do that.

**Comment by ksauraj** [comment]
You want this to be assigned under `IWOC2024`?

**Comment by Tholkappiar** [comment]
Yes ! Under IWOC 2024 !

**Comment by Tholkappiar** [comment]
> In my opinion, pdf will be better option. Also talking about csv it won't take much efforts as we already have option in `pandas` to do that. You can also go for both option ig.

Okay sure I will export in csv and in pdf too , i will also try exporting in json , is that okay for you ? consider the difficulty level for this feat issue , Thank you !

**Comment by ksauraj** [comment]
> > In my opinion, pdf will be better option. Also talking about csv it won't take much efforts as we already have option in `pandas` to do that. You can also go for both option ig.
> 
> Okay sure I will export in csv and in pdf too , i will also try exporting in json , is that okay for you ? consider the difficulty level for this feat issue , Thank you !

I hope you won't be using lot of libraries for that, if that's the case I might label it as `hard` considering you will add all three options.

**Comment by Tholkappiar** [comment]
yeah okay , i will take this as challenge and update you ! :+1:

**Comment by ksauraj** [comment]
> yeah okay , i will take this as challenge and update you ! 👍

Lol, All the best

**Comment by Tholkappiar** [comment]
Planning to include these libs in the script ,  is including these libs okay for the script ?
 I have to use the fpdf or some libs to export the pdf !
```
import json
from fpdf import FPDF
```

**Comment by Tholkappiar** [comment]
correctly exporting csv and json (also structured the json data while generating ), but exporting in pdf format is such a hard thing now , have any ideas regarding this , the text in the pdf is overlaying to other texts (used FPDF library) ?

**Comment by ksauraj** [comment]
> Planning to include these libs in the script , is including these libs okay for the script ? I have to use the fpdf or some libs to export the pdf !
> 
> ```
> import json
> from fpdf import FPDF
> ```

If no other option, use it.

**Comment by Tholkappiar** [comment]
> correctly exporting csv and json (also structured the json data while generating ), but exporting in pdf format is such a hard thing now , have any ideas regarding this , the text in the pdf is overlaying to other texts (used FPDF library) ?

Any ideas ?

**Comment by ksauraj** [comment]
> correctly exporting csv and json (also structured the json data while generating ), but exporting in pdf format is such a hard thing now , have any ideas regarding this , the text in the pdf is overlaying to other texts (used FPDF library) ?

Nah no idea about that, try other libs if available.

**Comment by aavvvacado** [comment]
I would love to try my hands on it , so could you please assign the given issue under hactoberfest 24 , also do suggests what could be possible approaches , i have read the above comments and thinking to working that in same manner

**Comment by ksauraj** [comment]
@aavvvacado thanks for showing your interest, can you please tell how you gonna do it ?
